src/hyp3_srg/utils.py

```diff
@@ -227,3 +228,54 @@ def how_many_gpus():
     (param, err) = proc.communicate()
     ngpus = int(str(param, 'UTF-8').split()[0])
     return ngpus
+
+
+def get_s3_args(uri: str, dest_dir: Optional[Path] = None) -> None:
+    """Retrieve the arguments for downloading from an S3 bucket
+
+    Args:
+        uri: URI of the file to download
+        dest_dir: the directory to place the downloaded file in
+
+    Returns:
+        bucket: the s3 bucket to download from
+        key: the path to the file following the s3 bucket
+        out_path: the destination path of the file to download
+    """
+    if dest_dir is None:
+        dest_dir = Path.cwd()
+
+    simple_s3_uri = Path(uri.replace('s3://', ''))
+    bucket = simple_s3_uri.parts[0]
+    key = '/'.join(simple_s3_uri.parts[1:])
+    out_path = dest_dir / simple_s3_uri.parts[-1]
+    return bucket, key, out_path
+
+
+def s3_list_objects(bucket: str, prefix: str = '') -> dict:
+    """List objects in bucket at prefix
+
+    Args:
+        bucket: the simple s3 bucket name
+        prefix: the path within the bucket to search
+
+    Returns:
+        res: dictionary containing the response
+    """
+    S3 = client('s3')
```

Comment:
We typically define `boto3` clients as global variables for performance reasons. I don't think it matters too much here, but it's good practice.